packages/core/src/data-layer.test.ts

```diff
@@ -0,0 +1,110 @@
+import {Store, Unsubscribe} from "./model";
+import {shallowEquals, subscribeWithSelector} from "./data-layer";
+import {createEmptyStore, identity, TestStore} from "./test-utils";
+
+describe("data-layer", () => {
+    describe("subscribeWithSelector", () => {
+        it("returns unsubscribe function", () => {
+            const store: Store<unknown> = createEmptyStore({})
+            const unsubscribe = jest.fn()
+            store.subscribe = jest.fn(() => unsubscribe)
+
+            expect(subscribeWithSelector(
+                store,
+                () => {},
+                () => {},
+                () => false
+            )).toBe(unsubscribe)
+        })
+        it("calls selector to get derived state", () => {
+            const state = {}, nextState = {a: 1}
```

Comment:
const {a: 1}

Reply:
what's wrong?

Reply:
Can you see const data?

Reply:
still don't understand

Reply:
but context is different every time. i don't know how to name this object if it's global - now it's named according to the context